Makefile

```diff
@@ -113,6 +113,69 @@ distclean:
 
 check: test
 
+# Remove files generated by running coverage, put the non-instrumented lib back 
+# in place 
+coverage-clean:
+	if [ -d lib_ ]; then rm -rf lib; mv lib_ lib; fi
+	-rm -rf gcovr
+	-rm -rf out/$(BUILDTYPE)/.coverage
+	-rm -rf .cov_tmp coverage
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/*.gcda
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/tracing/*.gcda
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/*.gcno
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/tracing*.gcno
+
+# Build and test with code coverage reporting.  Leave the lib directory 
+# instrumented for any additional runs the user may want to make.
+# For C++ coverage reporting, this needs to be run in conjunction with configure
+#  --coverage.  html coverage reports will be created under coverage/
+
+coverage: coverage-build coverage-test
+
+coverage-build: all
+	mkdir node_modules
+	$(NODE) ./deps/npm install istanbul-merge
+	$(NODE) ./deps/npm install nyc
+	git clone --depth=1 --single-branch git://github.com/gcovr/gcovr.git
+	git clone --depth=1 --single-branch https://github.com/nodejs/testing.git
+	(cd gcovr && patch -p1 < "$(CURDIR)/testing/coverage/gcovr-patches.diff")
+	rm -rf testing
+	mv lib lib_
+	$(NODE) ./node_modules/.bin/nyc instrument lib_/ lib/ 
+	$(MAKE)
+
+coverage-test:
+	$(MAKE) build-addons
+	$(MAKE) cctest
+	-rm -rf out/$(BUILDTYPE)/.coverage
+	-rm -rf .cov_tmp
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/*.gcda
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/tracing/*.gcda
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/*.gcno
+	-rm -f out/$(BUILDTYPE)/obj.target/node/src/tracing*.gcno
+	-$(PYTHON) tools/test.py --mode=release -J addons doctool inspector \
+		known_issues pseudo-tty parallel sequential
+	mv lib lib__ 
+	mv lib_ lib
```

Comment:
That looks like a major step forwards.  Does it need an existing node binary at the moment though - or different parameters to build.  When I tried out the build from your repo it failed in `out/Release/obj/gen/node_natives.h` because it could find the `node` command to run `node node_modules/.bin/nyc instrument lib/internal/bootstrap_node.js`